---
int32 newThreshold
int32 x
int32 y
int32 bestDark
